### PR TITLE
[bitnami/scylladb] Fix custom datacenter and rack value handling and config path directory

### LINKED ISSUE #74999: [bitnami/scylladb] Cannot change data center name with SCYLLADB_DATACENTER
### Name and Version

bitnami/scylladb

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Create a container based on the latest scylladb image and tweak the data center name with SCYLLADB_DATACENTER environment variable.
```
# docker run -d --name scylladb -e SCYLLADB_DATACENTER=dc1 bitnami/scylladb:latest
```
2. Get data center with cqlsh
```
# docker exec -it scylladb cqlsh --user cassandra --password cassandra

Warning: Using a password on the command line interface can be insecure.
Recommendation: use the credentials file to securely provide the password.

WARNING: cqlsh was built against 5.0.0, but this server is 3.0.8.  All features may not work!
Connected to My Cluster at 127.0.0.1:9042
[cqlsh 6.2.0 | Cassandra 3.0.8 | CQL spec 3.3.1 | Native protocol v4]
Use HELP for help.
cassandra@cqlsh> select data_center from system.local;

 data_center
-------------
 datacenter1

(1 rows)
```

### What is the expected behavior?

Expected to see `dc1 as the name for the local data center.

### What do you see instead?

The name is the default `datacenter1` value.

### Additional information

Looking at `cassandra-rackdc.properties` file inside the container, the value for `dc` is set correctly but the line is commented (same for `rack`). Seems like `cassandra_setup_rack_dc()` function in `libscylladb.sh` scripts doesn't remove comments from the line. One way maybe to remove leading `#` and spaces from the line:
```
diff --git a/bitnami/scylladb/6.0/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh b/bitnami/scylladb/6.0/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
index 7bf0faedc22..18987e6c589 100644
--- a/bitnami/scylladb/6.0/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
+++ b/bitnami/scylladb/6.0/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
@@ -797,8 +797,8 @@ cassandra_setup_common_ports() {
 #########################
 cassandra_setup_rack_dc() {
     if ! cassandra_is_file_external "${DB_MOUNTED_RACKDC_PATH}"; then
-        replace_in_file "${DB_RACKDC_FILE}" "dc=.*" "dc=${DB_DATACENTER}"
-        replace_in_file "${DB_RACKDC_FILE}" "rack=.*" "rack=${DB_RACK}"
+        replace_in_file "${DB_RACKDC_FILE}" "^(#\s)?(\s*)dc=.*" "dc=${DB_DATACENTER}"
+        replace_in_file "${DB_RACKDC_FILE}" "^(#\s)?(\s*)rack=.*" "rack=${DB_RACK}"
     else
         debug "${DB_MOUNTED_RACKDC_PATH} mounted. Skipping rack and datacenter configuration"
     fi
diff --git a/bitnami/scylladb/6.1/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh b/bitnami/scylladb/6.1/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
index 7bf0faedc22..18987e6c589 100644
--- a/bitnami/scylladb/6.1/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
+++ b/bitnami/scylladb/6.1/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
@@ -797,8 +797,8 @@ cassandra_setup_common_ports() {
 #########################
 cassandra_setup_rack_dc() {
     if ! cassandra_is_file_external "${DB_MOUNTED_RACKDC_PATH}"; then
-        replace_in_file "${DB_RACKDC_FILE}" "dc=.*" "dc=${DB_DATACENTER}"
-        replace_in_file "${DB_RACKDC_FILE}" "rack=.*" "rack=${DB_RACK}"
+        replace_in_file "${DB_RACKDC_FILE}" "^(#\s)?(\s*)dc=.*" "dc=${DB_DATACENTER}"
+        replace_in_file "${DB_RACKDC_FILE}" "^(#\s)?(\s*)rack=.*" "rack=${DB_RACK}"
     else
         debug "${DB_MOUNTED_RACKDC_PATH} mounted. Skipping rack and datacenter configuration"
     fi
diff --git a/bitnami/scylladb/6.2/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh b/bitnami/scylladb/6.2/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
index 7bf0faedc22..18987e6c589 100644
--- a/bitnami/scylladb/6.2/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
+++ b/bitnami/scylladb/6.2/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh
@@ -797,8 +797,8 @@ cassandra_setup_common_ports() {
 #########################
 cassandra_setup_rack_dc() {
     if ! cassandra_is_file_external "${DB_MOUNTED_RACKDC_PATH}"; then
-        replace_in_file "${DB_RACKDC_FILE}" "dc=.*" "dc=${DB_DATACENTER}"
-        replace_in_file "${DB_RACKDC_FILE}" "rack=.*" "rack=${DB_RACK}"
+        replace_in_file "${DB_RACKDC_FILE}" "^(#\s)?(\s*)dc=.*" "dc=${DB_DATACENTER}"
+        replace_in_file "${DB_RACKDC_FILE}" "^(#\s)?(\s*)rack=.*" "rack=${DB_RACK}"
     else
         debug "${DB_MOUNTED_RACKDC_PATH} mounted. Skipping rack and datacenter configuration"
     fi
```

### PR DESCRIPTION
**Pull request by siavasht**
### Description of the change

- Uncomment lines in `cassandra_setup_rack_dc` function in `libscylladb.sh`
- Export `SCYLLADB_CONF` environment variable in `scylladb-env.sh` to be same as `SCYLLADB_CONF_DIR`

### Benefits

- Allows for customizing data center name and rack from environment variables. Values were ignored previously because the bash function didn't remove comments.
- Fix startup failure when GossipingPropertyFileSnitch is selected. The failure happens because Scylladb tries to read `cassandra-rackdc.properties` file from an invalid path.

### Possible drawbacks

None that I know of.

### Applicable issues

- fixes #74999

### CODE REVIEW

File: bitnami/scylladb/6.2/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh [hunk 1]
```diff
@@ -797,8 +797,8 @@ cassandra_setup_common_ports() {
 #########################
 cassandra_setup_rack_dc() {
     if ! cassandra_is_file_external "${DB_MOUNTED_RACKDC_PATH}"; then
-        replace_in_file "${DB_RACKDC_FILE}" "dc=.*" "dc=${DB_DATACENTER}"
-        replace_in_file "${DB_RACKDC_FILE}" "rack=.*" "rack=${DB_RACK}"
+        replace_in_file "${DB_RACKDC_FILE}" "^(#\s)?(\s*)dc=.*" "dc=${DB_DATACENTER}"
+        replace_in_file "${DB_RACKDC_FILE}" "^(#\s)?(\s*)rack=.*" "rack=${DB_RACK}"
```
Comment (jotamartos) [inline bitnami/scylladb/6.2/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh:801]: What about simplifying the regex? 

```suggestion
        replace_in_file "${DB_RACKDC_FILE}" "^[#\s]*dc=.*" "dc=${DB_DATACENTER}"
        replace_in_file "${DB_RACKDC_FILE}" "^[#\s]*rack=.*" "rack=${DB_RACK}"
```

**Review by jotamartos — DISMISSED** [review]
Thank you for your contribution. Could you please take a look at my comment?

File: bitnami/scylladb/6.2/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh (hunk 1 above)
Comment (siavasht) [reply bitnami/scylladb/6.2/debian-12/rootfs/opt/bitnami/scripts/libscylladb.sh]: Looks good to me 🙂  I copied the expression from other functions in the script.

Updated the patch with your proposed change.

**Review by jotamartos — APPROVED** [review]
LGTM! We will push a new version of the container with the changes soon.

**Pull request merged**